src/interfaces/modules/IMarketplace.ts

```diff
@@ -0,0 +1,71 @@
+import { DirectListing } from "./../../types/marketplace/DirectListing";
+import { BigNumberish } from "ethers";
+
+import {
+  NewDirectListing,
+  DirectListing,
+  NewAuctionListing,
+  AuctionListing,
+  Offer,
+} from "../../types/marketplace";
+
+export interface IMarketplace {
+  /**
+   * Creates a new direct listing on a marketplace.
+   */
+  createDirectListing(listing: NewDirectListing): Promise<DirectListing>;
+
+  /**
+   * Creates a new auction listing on a marketplace.
+   */
+  createAuctionListing(listing: NewAuctionListing): Promise<AuctionListing>;
```

Comment:
I think we should return the ID rather than the listing same for ^.

Reply:
Yeah good call, updating.